core/modules/map.rs

```diff
@@ -1405,27 +1487,34 @@ impl ModuleMap {
       return Poll::Ready(Ok(()));
     }
 
+    self.prepare_buffered_dyn_imports();
+
     loop {
       let poll_result = self
         .preparing_dynamic_imports
         .borrow_mut()
         .poll_next_unpin(cx);
 
       if let Poll::Ready(Some(prepare_poll)) = poll_result {
-        let dyn_import_id = prepare_poll.0;
+        let dyn_imports = prepare_poll.0;
         let prepare_result = prepare_poll.1;
 
         match prepare_result {
-          Ok(load) => {
-            self
-              .pending_dynamic_imports
-              .borrow_mut()
-              .push(load.into_future());
+          Ok(()) => {
+            debug_assert!(!dyn_imports.is_empty());
+            for load in dyn_imports {
+              self
+                .pending_dynamic_imports
+                .borrow_mut()
+                .push(load.into_future());
+            }
             self.pending_dynamic_imports_pending.set(true);
           }
           Err(err) => {
             let exception = to_v8_type_error(scope, err);
-            self.dynamic_import_reject(scope, dyn_import_id, exception);
+            for load in dyn_imports {
+              self.dynamic_import_reject(scope, load.id, exception.clone());
```

Comment:
This is the part I'm most concerned about, since we just reject all of the import promises in this batch even though the error might only be caused by a subset.

Reply:
Let's change the return type of this hook to `Vec<Result<(), AnyError>>` 👍